src/ui-shell/sidenav/sidenav-item.component.ts

```diff
@@ -16,22 +17,37 @@ import { Router } from "@angular/router";
 @Component({
 	selector: "cds-sidenav-item, ibm-sidenav-item",
 	template: `
-		<a
-			class="cds--side-nav__link"
-			[ngClass]="{
+		<a *ngIf="!useRouter; else sidenavItemRouterTpl"
+		   class="cds--side-nav__link"
+		   [ngClass]="{
 				'cds--side-nav__item--active': active
 			}"
-			[href]="href"
-			[attr.aria-current]="(active ? 'page' : null)"
-			[attr.title]="title ? title : null"
-			(click)="navigate($event)">
+		   [href]="href"
+		   [attr.aria-current]="(active ? 'page' : null)"
+		   [attr.title]="title ? title : null"
+		   (click)="navigate($event)">
+			<ng-template [ngTemplateOutlet]="sidenavItemContentTpl"></ng-template>
+		</a>
+
+		<ng-template #sidenavItemRouterTpl>
+			<a
+				[routerLink]="route"
+				[routerLinkActive]="'cds--side-nav__item--active'"
+				[ariaCurrentWhenActive]="'page'"
+				[attr.title]="title ? title : null"
+				class="cds--side-nav__link">
```

Comment:
I don't think we need to bind routerlinkactive & ariaCurrentWhenActive here. We should be able to just do:

```suggestion
				[routerLink]="route"
				routerLinkActive="cds--side-nav__item--active"
				ariaCurrentWhenActive="page"
				[attr.title]="title ? title : null"
				class="cds--side-nav__link">
```

Reply:
You're absolutely right